### ISSUE TITLE
[wallee]: 

### ISSUE THREAD

**Issue by 1204Yoshi** [ghostery/trackerdb#266]
### Request

https://www.google.com/recaptcha/api.js

### Location

https://www.redgirasol.com/

### Tracker Company

Google

### Company Website

https://www.google.com/

### Category

Site Analytics

### Request Type

None

**Comment by philipp-classen** [comment]
With Ghostery 8, I see Google Analytics and Recaptcha requests listed under "wallee":

![wallee](https://github.com/ghostery/trackerdb/assets/1616509/87379da6-0582-4ae6-96cb-dcaab59ad948)

My expectation would have been that they are listed under Google Analytics and und Recaptcha. How, they match "wallee" is not clear to me. The entry is her: https://github.com/ghostery/trackerdb/blob/3ca21ccc9947bc1a7b6756ba912d7a8a2a4f8956/db/patterns/wallee.eno#L7

It only contains the domain "app-wallee.com", but I none of the requests have that domain.

**Comment by 1204Yoshi** [comment]
![image](https://github.com/ghostery/trackerdb/assets/164535032/0e13965c-4034-400e-9272-d333bf7f6dc1)

**Comment by philipp-classen** [comment]
Thanks for finding that! It looks like there is a bug in our matching logic. We need to investigate.

**Comment by philipp-classen** [comment]
The trackerdb API (`@ghostery/trackerdb@1.0.121`) attributes it correctly:

```
npx @ghostery/trackerdb "https://region1.analytics.google.com/g/collect?v=2&tid=G-QLYB1PK7RN&_ng=1&gtm=45je44f0v885777672z8894232443za200&_p=1713378081823&gcd=13l3l3l2l1&npa=1&dma_cps=sypham&dma=1&cid=1609738265.1713374177&ul=en-us&sr=1920x1200&uaa=x86&uab=64&uafvl=Not-A.Brand%3B99.0.0.0%7CChromium%3B124.0.6367.49&uamb=0&uam=&uap=Linux&uapv=6.8.4&uaw=0&pscdl=noapi&_s=1&sid=1713377776&sct=2&seg=1&dl=https%3A%2F%2Fwww.redgirasol.com%2Frendimiento&dt=RedGirasol&en=page_view&tfd=679"
```
->
```
 {
  "url": "https://region1.analytics.google.com/g/collect?v=2&tid=G-QLYB1PK7RN&_ng=1&gtm=45je44f0v885777672z8894232443za200&_p=1713378081823&gcd=13l3l3l2l1&npa=1&dma_cps=sypham&dma=1&cid=1609738265.1713374177&ul=en-us&sr=1920x1200&uaa=x86&uab=64&uafvl=Not-A.Brand%3B99.0.0.0%7CChromium%3B124.0.6367.49&uamb=0&uam=&uap=Linux&uapv=6.8.4&uaw=0&pscdl=noapi&_s=1&sid=1713377776&sct=2&seg=1&dl=https%3A%2F%2Fwww.redgirasol.com%2Frendimiento&dt=RedGirasol&en=page_view&tfd=679",
  "matches": [
    {
      "pattern": {
        "key": "google_signals",
        "name": "Google Analytics with signals",
        "category": "advertising",
        "organization": "google",
        "alias": null,
        "website_url": "https://support.google.com/analytics/answer/9445345?hl=en#zippy=%2Cin-this-article",
        "ghostery_id": null,
        "domains": [
          "analytics.google.com"
        ],
        "filters": [
          "||analytics.google.com^$third-party",
          "||region1.analytics.google.com^$third-party"
        ]
      },
      "category": {
        "key": "advertising",
        "name": "Advertising",
        "color": "#cb55cd",
        "description": "Advertising services that utilize data collection, behavioral analysis, and user retargeting."
      },
      "organization": {
        "key": "google",
        "name": "Google",
        "description": "Google LLC is an American multinational technology company that specializes in Internet-related services and products, which include online advertising technologies, search engine, cloud computing, software, and hardware.",
        "website_url": "https://www.google.com",
        "country": "US",
        "privacy_policy_url": "https://www.google.com/intl/en/policies/privacy/",
        "privacy_contact": "https://www.google.com/contact/",
        "ghostery_id": "82"
      }
    }
  ]
}
```

And also

```
npx @ghostery/trackerdb "https://www.gstatic.com/recaptcha/releases/rz4DvU-cY2JYCwHSTck0_qm-/recaptcha__en.js"
```
->
```
{
  "url": "https://www.gstatic.com/recaptcha/releases/rz4DvU-cY2JYCwHSTck0_qm-/recaptcha__en.js",
  "matches": [
    {
      "pattern": {
        "key": "google_recaptcha",
        "name": "Google Recaptcha",
        "category": "essential",
        "organization": "google",
        "alias": null,
        "website_url": "https://www.google.com/recaptcha/about/",
        "ghostery_id": null,
        "domains": [
          "recaptcha.net"
        ],
        "filters": [
          "||recaptcha.net/recaptcha/",
          "||google.com/recaptcha/$frame,script",
          "||gstatic.*/recaptcha/"
        ]
      },
      "category": {
        "key": "essential",
        "name": "Essential",
        "color": "#fc9734",
        "description": "Site requests that may be critical to website functionality, such as tag managers and privacy notices."
      },
      "organization": {
        "key": "google",
        "name": "Google",
        "description": "Google LLC is an American multinational technology company that specializes in Internet-related services and products, which include online advertising technologies, search engine, cloud computing, software, and hardware.",
        "website_url": "https://www.google.com",
        "country": "US",
        "privacy_policy_url": "https://www.google.com/intl/en/policies/privacy/",
        "privacy_contact": "https://www.google.com/contact/",
        "ghostery_id": "82"
      }
    },
    {
      "pattern": {
        "key": "gstatic",
        "name": "Google Static",
        "category": "cdn",
        "organization": "google",
        "alias": null,
        "website_url": null,
        "ghostery_id": "3588",
        "domains": [
          "gstatic.com"
        ],
        "filters": []
      },
      "category": {
        "key": "cdn",
        "name": "CDN",
        "color": "#4ecba1",
        "description": "Content Delivery Networks (CDNs) distribute resources for different site utilities, and collect user analytics and tracking data."
      },
      "organization": {
        "key": "google",
        "name": "Google",
        "description": "Google LLC is an American multinational technology company that specializes in Internet-related services and products, which include online advertising technologies, search engine, cloud computing, software, and hardware.",
        "website_url": "https://www.google.com",
        "country": "US",
        "privacy_policy_url": "https://www.google.com/intl/en/policies/privacy/",
        "privacy_contact": "https://www.google.com/contact/",
        "ghostery_id": "82"
      }
    }
  ]
}
```

Yet in the Ghostery Extensions, the requests ended up being listed as `wallee`.

**Comment by philipp-classen** [comment]
A fix should come with the next release of Ghostery (https://github.com/ghostery/ghostery-extension/pull/1544). I can confirm locally, that it fixes the problem for me.

**Comment by cookiefirst** [comment]
Our consent management software Called CookieFirst has also been classified als Walleee and not as the tracker that we have submitted [before](https://github.com/ghostery/trackerdb/issues/219)

![CleanShot 2024-05-01 at 13 17 04@2x](https://github.com/ghostery/trackerdb/assets/93817342/038f2217-9719-4185-98ac-da1540f3f1fc)

**Comment by philipp-classen** [comment]
@cookiefirst Indeed, we need to look into it again. On a production profile, I also still see "wallee".

Either the fix is not released yet, or it is not working for a different reason.

**Comment by philipp-classen** [comment]
Hit us now also in another way: recaptcha was missclassified as "wallee" and thus started to get blocked, causing more sites to break.

The concrete case of recaptcha being blocked should be solved now by https://github.com/ghostery/broken-page-reports/pull/624 and https://github.com/ghostery/trackerdb/pull/283. But their may be other unintended side-effects of the misclassification.

**Comment by philipp-classen** [comment]
The fix is included in the upcoming Ghostery release 8.12.10 (https://github.com/ghostery/ghostery-extension/pull/1576).

**Comment by 1204Yoshi** [comment]
![image](https://github.com/ghostery/trackerdb/assets/164535032/63e084e2-f59e-46e0-89bb-acf854bf933f)

Maybe depend on the kind of site

**Comment by chrmod** [comment]
Thanks to your report, we have successfully identified and fixed the problem with incorrect attribution of requests to wrong trackers. 
The fix is on a way as a part of Ghostery [8.12.10](https://github.com/ghostery/ghostery-extension/releases/tag/v8.12.10)

Thank you for all the effort you've put to help improve Ghostery, it is much appreciated!

**Comment by philipp-classen** [comment]
Firefox has been also updated now.
